Add a page on debugging Ember CLI

mehulkar opened the pull request:
Closes #151

Took a stab at the content here and approached it as something useful for beginner frontend developers.

jenweber commented on guides/advanced-use/debugging.md:
```diff
@@ -0,0 +1,80 @@
+a LiveReload development server at `http://localhost:4200`. Run `ember build`, and you get a `dist/`
+directory with compiled assets ready to be deployed to your production server.
+
+But things don't always go smoothly and CLI commands can fail inexplicably with error messages that are
+difficult to understand or act upon. In these cases, it can be useful to have tools to figure out what
```
```suggestion
But things don't always go smoothly and CLI commands can fail inexplicably with error messages that
can sometimes be difficult to act on. In these cases, it can be useful to have tools to figure out what
```

mehulkar commented:
@jenweber thanks for the quick review! Added your suggestion. Not sure what the convention is for this repo, but I can squash the commits if you prefer also.

jenweber commented on guides/advanced-use/debugging.md:
````diff
@@ -0,0 +1,80 @@
+DEBUG=ember-cli:* ember build
+```
+
+If you set the variable to `*`. Consult the `debug` library documentation for additional ways to
````
```suggestion
You can set the variable to `*`. Consult the `debug` library documentation for additional ways to
```

jenweber commented on guides/advanced-use/debugging.md:
```diff
@@ -0,0 +1,80 @@
+
+## Linking Addons
+
+While logging and breakpoints are useful, there are additional challenges when the code you're trying
```
```suggestion
While logging and breakpoints are useful, there are additional challenges when the code you are trying
```

This suggestion is because generally we try to limit contractions in order to make our docs easier for non-native English speakers.

jenweber commented on guides/advanced-use/debugging.md:
```diff
@@ -0,0 +1,80 @@
+1. In the addon repo, run `npm link`
+1. In your app, run `npm link ember-power-select`
+
+You can verify this did the intended thing by checking id `node_modules/ember-power-select` is now
```
```suggestion
You can verify this did the intended thing by checking that `node_modules/ember-power-select` is now
```

jenweber commented:
If you want to squash, go for it. It's not required.

mehulkar commented:
One other thing, I saw there's a Debugging section in Assets & Dependencies. I can either add a Read More style link there, or I can move that content to this new page. What do you think?

https://cli.emberjs.com/release/basic-use/assets-and-dependencies/#debuggingnode_modulesdependencies

maxwondercorn commented:
There is also debugging info on node and blueprints here

https://cli.emberjs.com/release/appendix/dev-tools/#debuggingnodecode

jenweber commented:
I think move the main content to your debugging article, but keep the intro where it is. Something like this should stay on `assets and dependencies`:

## Debugging node_modules dependencies

Errors such as "a module named _____ could not be found" or a colleague's report that "well, the app works on my computer but not yours!" sometimes indicate that the local server needs to be restarted or node_modules should be reinstalled.

To overcome issues like this, visit [the CLI debugging guide]().

jenweber commented:
Developer tools could have similar treatment as ^^^

mehulkar commented:
@maxwondercorn ooh nice callout, thanks! I moved over both and added links. Let me know. I felt the "debugging custom blueprints" was a bit repetitive and the "Breakpoints" section covers it more generally. Let me know what you think
